react/SearchWrapper.tsx

```diff
@@ -115,6 +116,75 @@ const getSearchIdentifier = (
   return query + map + (orderBy ?? '') + (page ?? '')
 }
 
+function getCanonicalMaybeWithPage(canonicalLink: string, page: number) {
+  if (page <= 0) {
+    return null
+  }
+
+  const canonicalWithPage = `${canonicalLink}?page=${page}`
+
+  return canonicalWithPage
+}
+
+function getHelmetLink(
+  canonicalLink: string | null,
+  page: number,
+  rel: 'canonical' | 'prev' | 'next'
+) {
+  if (!canonicalLink) {
+    return null
+  }
+
+  let pageAfterTransformation = 0
+
+  if (rel === 'canonical') {
+    pageAfterTransformation = page
+  } else if (rel === 'next') {
+    pageAfterTransformation = page + 1
+  } else if (rel === 'prev') {
+    pageAfterTransformation = page - 1
+  }
+
+  if (pageAfterTransformation === 1) {
+    return {
+      rel,
+      href: encodeURI(canonicalLink),
+    }
+  }
+
+  const canonicalMaybeWithPage = getCanonicalMaybeWithPage(
+    canonicalLink,
+    pageAfterTransformation
+  )
+
+  if (!canonicalMaybeWithPage) {
+    return null
+  }
+
+  return {
+    rel,
+    href: encodeURI(canonicalMaybeWithPage),
+  }
+}
+
+function isNotLastPage(
+  products: SearchQuery['products'],
+  to: number | undefined,
+  recordsFiltered: number | undefined
+) {
+  if (
+    !products ||
+    to === undefined ||
+    to === null ||
+    recordsFiltered === undefined ||
+    recordsFiltered === null
+  ) {
+    return null
+  }
+
+  return to + 1 < recordsFiltered
```

Comment:
This was taken from [here](https://github.com/vtex-apps/search-result/blob/eaefcf20c6953ef31c4f69724a96c48e7e0e2e78/react/components/loaders/FetchMoreButton.js#L18) 😄